### ISSUE TITLE
Uncaught TypeError: Cannot read property 'getSelectionComponent' of null

### ISSUE THREAD

**Issue by domoritz** [vega/vega-lite#2635]
<img width="437" alt="screen shot 2017-07-06 at 10 27 22" src="https://user-images.githubusercontent.com/589034/27924331-c5bd2e0a-6235-11e7-867e-9e2e6b8dbf91.png">

```json
{
  "$schema": "https://vega.github.io/schema/vega-lite/v2.json",
  "data": {"url": "data/europe_world_bank.csv"},
  "hconcat": [
    {
      "repeat": {
        "column": ["Population_ages_15_64_of_total", "Population_ages_65_and_above_of_total"]
      },
      "spec": {
        "mark": "bar",
        "height": 700,
        "selection": {"brush": {"type": "interval", "encodings": ["y"]}},
        "encoding": {
          "y": {
            "field": "Country",
            "type": "ordinal",
            "sort": {
              "op": "min",
              "field": "Population_ages_15_64_of_total",
              "order": "descending"
            }
          },
          "x": {
            "field": "Population_ages_15_64_of_total",
            "type": "quantitative"
          },
          "color": {
            "condition": {"selection": "!brush", "value": "steelblue"},
            "value": "goldenrod"
          }
        }
      }
    },
    {
      "mark": "point",
      "width": 260,
      "height": 260,
      "selection": {"brush": {"type": "interval"}},
      "encoding": {
        "y": {
          "field": "Population_ages_15_64_of_total",
          "type": "quantitative",
          "scale": {"zero": false}
        },
        "x": {
          "field": "Population_ages_65_and_above_of_total",
          "type": "quantitative",
          "scale": {"zero": false}
        },
        "color": {
          "condition": {"selection": "!brush", "value": "steelblue"},
          "value": "goldenrod"
        }
      }
    }
  ]
}
```

**Comment by arvind** [comment]
With #2471, the `!name` syntax is no longer supported. Instead, try:

```json
"condition": {"selection": {"not": "brush"}, "value": "steelblue"}
```

With that said, we should throw a better error here.

**Comment by domoritz** [comment]
Thanks!

Should selections over ordinal scales work?

![editor](https://user-images.githubusercontent.com/589034/27935127-faea1d22-625d-11e7-8955-d47573e2bfbe.gif)

**Comment by arvind** [comment]
@domoritz: Should be fixed with #2639. Sorry for the regression.
